resource-aggregate/cqrs/eventstore/eventstore.go

```diff
@@ -13,16 +13,28 @@ type VersionQuery struct {
 
 // SnapshotQuery used to load events from snapshot.
 type SnapshotQuery struct {
-	GroupID           string //filter by group ID
-	AggregateID       string //filter to certain aggregateID, groupID is required
-	SnapshotEventType string //required
+	GroupID     string //filter by group ID
+	AggregateID string //filter to certain aggregateID, groupID is required
 }
 
+type SaveStatus int
+
+const (
+	OK                   SaveStatus = 0  // events were stored
+	ConcurrencyException SaveStatus = 1  // need to reload events from store, because the versions of events are obsolete by concurrent store.
+	NeedSnapshot         SaveStatus = 2  // the first event need to be snapshot, because store for aggregate is exhausted.
+	FAIL                 SaveStatus = -1 // error occurs
```

Comment:
Why all capitals? Fail would be consistent with others.

Reply:
error occur**ed**

Reply:
fixed